### PR TITLE
[SPARK-19674][SQL]Ignore driver accumulator updates don't belong to the execution when merging all accumulator updates

### PR DESCRIPTION
**Pull request by carsonwang**
## What changes were proposed in this pull request?
In SQLListener.getExecutionMetrics, driver accumulator updates don't belong to the execution should be ignored when merging all accumulator updates to prevent NoSuchElementException.

## How was this patch tested?
Updated unit test.

### CODE REVIEW

**Comment by carsonwang** [comment]
cc @cloud-fan @zsxwing

File: sql/core/src/test/scala/org/apache/spark/sql/execution/ui/SQLListenerSuite.scala [hunk 1]
```diff
@@ -147,6 +147,10 @@ class SQLListenerSuite extends SparkFunSuite with SharedSQLContext with JsonTest
 
     checkAnswer(listener.getExecutionMetrics(0), accumulatorUpdates.mapValues(_ * 2))
 
+    // Non-existing driver accumulator updates should be filtered and no exception will be thrown.
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/ui/SQLListenerSuite.scala]: when will this happen?

**Comment by cloud-fan** [comment]
The change looks reasonable

File: sql/core/src/test/scala/org/apache/spark/sql/execution/ui/SQLListenerSuite.scala (hunk 1 above)
Comment (carsonwang) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/ui/SQLListenerSuite.scala]: I was doing some own experiments that adds physical operators at runtime. The metrics of the added operators are not registered to the execution so I got a NoSuchElementException.

**Comment by carsonwang** [comment]
Thanks @cloud-fan . `driver accumulators don't belong to this execution` is more appropriate. I'll update the words.

**Comment by cloud-fan** [comment]
retest this please

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Comment by mallman** [comment]
It looks like this will fix a bug we're experiencing in Spark 2.1. Given that this PR is a bug fix, any chance we can get a backport into `branch-2.1`? I can work on it myself if @carsonwang is unable.

**Comment by mallman** [comment]
I created a Spark 2.1 backport at #17418.

**Pull request closed without merging**